### PR TITLE
Adds a recoilless rifle, bag to carry the ammo and normal rockets

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Adds a recoilless rifle to marines, and it is in SL vendors (probably subject to change) (45 points) and 40 points in req as a kit. 
The kit has a rocket bag (still need a different sprite) which itself holds 5 shells/rockets. The shells are too big to fit elsewhere.
The rifle cannot be held anywhere else other than your suit storage.
It has LE (big small explosion over a big area, goes offscreen like a normal bullet) and HE (Big explosion over a small area, works like normal rockets currently in game)
Needs bag sprites and its done.

## Why It's Good For The Game
More early game options for marines, SLs need a bit of love for now and this might be a nice change.

## Changelog
:cl: MJP, Kuro
add: Added a recoilless rifle and bag for it. Added it to SL vendors. Added it to req.
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -42. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

File: code/modules/projectiles/ammo_datums.dm
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1106]:
```suggestion
	hud_state_empty = "shell_empty"
```

File: code/modules/projectiles/ammo_datums.dm
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1127]:
```suggestion
```

File: code/modules/projectiles/ammo_datums.dm
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1148]:
```suggestion
```

File: code/modules/projectiles/guns/specialist.dm
Comment (TiviPlus) [inline code/modules/projectiles/guns/specialist.dm:959]: better type name please, for example /recoilless_rifle

File: code/modules/projectiles/guns/specialist.dm
Comment (TiviPlus) [inline code/modules/projectiles/guns/specialist.dm:985]:
```suggestion
```
smh at least check your copypaste

**Review by TiviPlus — COMMENTED** [review]
Will review again later

**Comment by Hathkar** [comment]
Looks a bit too much like an RPG when on the back and wielded.

**Comment by Chaznoodles** [comment]
Seems weird that the shells don't fit in G8 storage or backpacks/satchels, whereas you can fit whole mortar shells or big RPG rockets in them.

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Make it either a standard squad leader equipment that makes you worse at using it the more leadership you have (ideal option) or make it req only.
SLs using all their points to be wannabe specs is lame

**Review by TiviPlus — APPROVED** [review]
TM for a few rounds to see how putting it in req plays out, will merge after

**Review by TiviPlus — APPROVED** [review]

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-45). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**
